aiida/tools/dbimporters/plugins/matp.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module that contains the class definitions necessary to offer support for
+queries to Materials Project."""
+
+from __future__ import absolute_import
+import os
+import datetime
+import requests
+
+from pymatgen import MPRester
+from aiida.tools.dbimporters.baseclasses import CifEntry, DbEntry, DbImporter, DbSearchResults
+
+
+class MatProjImporter(DbImporter):
+    """
+    Database importer for the Materials Project.
+    """
+
+    _collection = 'structures'
+    _supported_keywords = None
+
+    def __init__(self, api_key=None):
+        """
+        Instantiate the MatProjImporter by setting up the Materials API (MAPI) connection details
+
+        :param api_key: the API key to be used to access MAPI
+        """
+        self.setup_db(api_key=api_key)
+        self._mpr = MPRester(self._api_key)
+
+    def setup_db(self, **kwargs):
+        """
+        Setup the required parameters to the REST API
+
+        :param api_key: the API key to be used to access MAPI
+        """
+        api_key = kwargs['api_key']
+        if api_key is None:
+            try:
+                self._api_key = os.environ['PMG_MAPI_KEY']
```

Comment:
Not sure whether it is safe to encourage users to put the private key in OS environment, as:

* environment might get stored in Calculation objects and exported in AiiDA database dumps;
* there might be other programs that read and save the environment, proliferating the key further.

Probably better is to use AiiDA properties (for the list of currently defined properties see ``verdi devel describeproperties``).

Reply:
True, but Materials Project use `PMG_MAPI_KEY`. If a user already follows that, I believe Aiida could make use of that. If the environment gets stored, the user might have all sorts of thing in there that one really should not spread around. Aiida already have this in the MPDS plugin. That is also why I just did the same. That is not in `master`, so this is something we can change. Should we remove it from both?

Reply:
OK, I was not aware that ``PMG_MAPI_KEY`` environment variables are already used by Materials Project and other software. Then it makes no sense to invent our own way. Anyway we should make sure that ``PMG_MAPI_KEY`` (and other special variables) do not get stored in AiiDA database.